### PR TITLE
Change list of buttons to checkbox for filtering

### PR DESCRIPTION
**Pull request by davidhao3300**
We'll need to humanize the club names later.
screen shot 2015-04-17 at 2 57 08 pm

### CODE REVIEW

File: cal/__init__.py [hunk 1]
```diff
@@ -88,7 +88,9 @@ def users():
                         .filter(Event.start < now + dt.timedelta(weeks=1))
 
     users = {event.user for event in events}    # use set to make users unique
-    return jsonify(data=[user.to_json() for user in users])
+    users = [user.to_json() for user in sorted(list(users),  # sort by name
+                                               key=lambda user: user.name)]
```
Comment (alanhdu) [inline cal/__init__.py]: Could we do (one line, no `list(users)` call).

``` Python
users = [user.to_json() for user in sorted(users, key=lambda user: user.name)]
```

If that's too long for pep8, we can do

``` Python
users = [user.to_json() for user in sorted(users, key=lambda u: u.name)]
```

File: cal/static/jsx/query.jsx [hunk 2]
```diff
@@ -19,6 +19,7 @@ var RSearch = React.createClass({
 
 var RReset = React.createClass({
     onclick: function() {
+        $("input[name='user'").prop('checked', true);
```
Comment (alanhdu) [inline cal/static/jsx/query.jsx:23]: Huh. Didn't know you could do that in jquery. What does passing in `true` do? Does it toggle, or set to `true`?

File: cal/static/jsx/query.jsx [hunk 3]
```diff
@@ -36,24 +37,52 @@ var RReset = React.createClass({
     }
 })
 
-var RFiltering = React.createClass({
+var RFilterCheckbox = React.createClass({
+    render: function() {
+        var user = this.props.user;
+        return (
+            <div className="filterForm">
+                <input type="checkbox" name="user" id={user.id} onChange={this.handleChange} defaultChecked="true"/>
```
Comment (alanhdu) [inline cal/static/jsx/query.jsx]: Could you break this up into multiple lines? It's a little long for my screen.

File: cal/static/jsx/query.jsx [hunk 4]
```diff
@@ -36,24 +37,52 @@ var RReset = React.createClass({
...
+    },
+
+    handleChange: function(e) {
+        // If user is checked, add events.
+        if (e.target.checked) {
+            this.props.add(e.target.id);
+        }
+        else {
+            this.props.remove(e.target.id);
+        }
```
Comment (alanhdu) [inline cal/static/jsx/query.jsx:61]: I don't think this is very `react`-esque. `this.props` is supposed to immutable. See [this](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md).

If we want to add a user_id, we should change `state`, and let React propagate the changes.

File: cal/static/jsx/query.jsx [hunk 5]
```diff
@@ -36,24 +37,52 @@ var RReset = React.createClass({
...
         this.props.removeUser(uid)
     },
 
-    render: function() {
-        var users = [];
-        for (var i = 0; i < this.props.userList.length; i++) {
-            var user = this.props.userList[i];
+    add: function(uid) {
+        this.props.addUser(uid)
+    },
```
Comment (alanhdu) [inline cal/static/jsx/query.jsx]: Again, `this.props` should be immutable.

File: cal/__init__.py (hunk 1 above)
Comment (davidhao3300) [inline cal/__init__.py]: Right, I didn't know sets could be sorted as well.

File: cal/static/jsx/query.jsx (hunk 2 above)
Comment (davidhao3300) [inline cal/static/jsx/query.jsx:23]: It sets the property checked to true.

File: cal/static/jsx/query.jsx (hunk 4 above)
Comment (davidhao3300) [inline cal/static/jsx/query.jsx:61]: Add and remove are not actually modifying this.props; I'll rename the methods so that it's not as confusing.

**Comment by alanhdu** [comment]
LGTM. Merging.

**Pull request merged**